app/jobs/hyrax/grant_edit_job.rb

```diff
@@ -5,10 +5,18 @@ class GrantEditJob < ApplicationJob
 
     # @param [String] file_set_id - the identifier of the object to grant access to
     # @param [String] user_key - the user to add
-    def perform(file_set_id, user_key)
-      file_set = ::FileSet.find(file_set_id)
-      file_set.edit_users += [user_key]
-      file_set.save!
+    def perform(file_set_id, user_key, use_valkyrie: Hyrax.config.use_valkyrie?)
+      if use_valkyrie
+        id = Valkyrie::ID.new(file_set_id)
+        file_set_resource = Hyrax.query_service.find_by(id: id)
+        permissions = PermissionManager.new(resource: file_set_resource)
```

Comment:
This looks good.

There's also a grant DSL on `Hyrax::AccessControlList`:

```ruby
file_set = Hyrax.query_service.find_by(id: file_set_id)
acl        = Hyrax::AccessControlList.new(resource: file_set)

acl.grant(:edit).to(User.find_by_user_key(user_key))
acl.save
```
I'll leave it up to you whether that DSL is attractive here.